### PR TITLE
feat: context (TresContext) is now exposed from TresCanvas

### PR DESCRIPTION
**Pull request by Tinoooo**

### CODE REVIEW

**Review by alvarosabu — APPROVED** [review]

**Comment by alvarosabu** [comment]
@Tinoooo should we add this to the docs?

**Comment by Tinoooo** [comment]
> @Tinoooo should we add this to the docs?

Yes! I forgot. I'm going to add the content now.

File: docs/api/composables.md [hunk 1]
```diff
@@ -136,12 +136,12 @@ Then you can bind the textures to the material.
       <TresSphereGeometry />
       <TresMeshStandardMaterial
         :map="map"
-        :displacementMap="displacementMap"
-        :normalMap="normalMap"
-        :roughnessMap="roughnessMap"
-        :metalnessMap="metalnessMap"
-        :aoMap="aoMap"
-        :alphaMap="alphaMap"
+        :displacement-map="displacementMap"
```
Comment (JaimeTorrealba) [inline docs/api/composables.md:139]: This will break the property?

**Review by JaimeTorrealba — COMMENTED** [review]
The addition of the tres-canvas page is a nice one!!

File: docs/api/composables.md (hunk 1 above)
Comment (Tinoooo) [reply docs/api/composables.md:139]: I don't think so. It is not a nested property (https://threejs.org/docs/index.html#api/en/materials/MeshStandardMaterial.displacementMap).

File: docs/api/composables.md [hunk 2]
```diff
@@ -179,19 +179,17 @@ watch(carRef, ({ model }) => {
 })
 ```
 
-## useTresContext (former useTres)
-
-This composable aims to provide access to the state model which contains the default renderer, camera, scene, and other useful properties.
+## useTresContext
+<!-- IMPROVE -->
```
Comment (JaimeTorrealba) [inline docs/api/composables.md:183]: We could remove this right?

File: docs/api/composables.md [hunk 3]
```diff
@@ -179,19 +179,17 @@ watch(carRef, ({ model }) => {
...
+This composable aims to provide access to the state model which contains multiple useful properties.
 
 ```ts
-const { camera, renderer } = useTresContext()
+const { camera, renderer, camera, cameras } = useTresContext()
```
Comment (JaimeTorrealba) [inline docs/api/composables.md:187]: And here we have two cameras, the rest looks OK to me

File: docs/api/composables.md [hunk 4]
```diff
@@ -209,3 +207,19 @@ import { useTresContext } from '@tresjs/core'
...
 ```
+
+### Properties of context
+| Property | Description |
+| --- | --- |
+| **camera** | the currently active camera |
+| **cameras** | the cameras that exist in the scene |
+| **controls** | the controls of your scene |
+| **deregisterCamera** | a method to deregister a camera. This is only required if you manually create a camera. Cameras in the template are deregistered automatically. |
+| **extend** | extends the component catalogue|
```
Comment (JaimeTorrealba) [inline docs/api/composables.md:218]: Here we can reference the extend docs https://tresjs.org/advanced/extending.html

**Review by JaimeTorrealba — APPROVED** [review]
I wrote some comments, but I'll approve, thanks for the docs, It's really useful for me :)

**Comment by alvarosabu** [comment]
I like the new structure @Tinoooo very well done

File: docs/api/composables.md (hunk 2 above)
Comment (Tinoooo) [reply docs/api/composables.md:183]: Yes. I removed it.

File: docs/api/composables.md (hunk 3 above)
Comment (Tinoooo) [reply docs/api/composables.md:187]: camera is the active camera and camera**s** is the array of existing cameras.

File: docs/api/composables.md (hunk 4 above)
Comment (Tinoooo) [reply docs/api/composables.md]: I added that.

File: docs/api/composables.md (hunk 3 above)
Comment (JaimeTorrealba) [reply docs/api/composables.md:186]: Yeah but looks, it says, camera, render CAMERA, cameras, is that ok?

**Review by alvarosabu — APPROVED** [review]

**Pull request merged**